flow/project.py

```diff
@@ -2054,13 +2043,103 @@ def _get_cached_status(self):
         except KeyError:
             return {}
 
-    def _get_operations_status(self, jobs, cached_status):
-        """Return a dict with information about job-operations for this aggregate.
+    def _generate_selected_aggregate_groups(
+        self,
+        selected_aggregates=None,
+        selected_groups=None,
+        tqdm_kwargs=None,
+    ):
+        """Yield selected aggregates and groups.
 
         Parameters
         ----------
-        jobs : :class:`~signac.contrib.job.Job` or aggregate of jobs
-            The signac job or aggregate.
+        selected_aggregates : iterable of aggregates
+            Aggregates to select.
+        selected_groups : set of :class:`~.FlowGroup`
+            Groups to select.
+        tqdm_kwargs : dict or None
+            A dict of keyword arguments to the tqdm progress bar used for
+            iterating over aggregates. If None, no progress bar will be
+            shown. (Default value = None)
+
+        Yields
+        ------
+        aggregate_id : str
+            Selected aggregate id.
+        aggregate : tuple of :class:`~signac.contrib.job.Job`
+            Selected aggregate.
+        group : :class:`~.FlowGroup`
+            Selected group.
+
+        """
+
+        def aggregate_progress_wrapper(aggregates):
+            """Show progress bar if keyword arguments are provided."""
+            if tqdm_kwargs is not None:
+                return tqdm(
+                    aggregates,
+                    total=len(aggregates),
+                    **tqdm_kwargs,
+                )
+            else:
+                return aggregates
+
+        if (
+            selected_groups is not None
+            and len(selected_groups) >= 0
+            and len(self._group_to_aggregate_store.inverse) > 1
+        ):
+            # Use only aggregate stores for the selected groups.
+            selected_aggregate_stores = {
+                self._group_to_aggregate_store[group] for group in selected_groups
+            }
+            aggregate_stores = {
+                aggregate_store: self._group_to_aggregate_store.inverse[aggregate_store]
+                for aggregate_store in selected_aggregate_stores
+            }
```

Comment:
I'm a little confused, maybe because of naming. Is `selected_aggregate_stores` just the set of aggregates from the selected groups? and then `aggregate_stores` is the dict that maps these aggregates to their groups?

Reply:
Good question. You're mostly correct about the descriptions of both variables. To clarify the first one, I might say `selected_aggregate_stores` is the set of _aggregate stores_ (not the set of _aggregates_) that correspond to the selected groups. I agree this is a potentially confusing name. I will rename `selected_aggregate_stores` to `aggregate_stores_of_selected_groups` to try and make it clearer.

Reply:
great, thanks! having `x` and `selected_x` implies to me that theyre the same type and one is a subset of the other.